pkg/flow/internal/worker/worker_pool.go

```diff
@@ -0,0 +1,125 @@
+package worker
+
+import (
+	"fmt"
+	"hash/fnv"
+	"math/rand"
+	"runtime"
+	"sync"
+)
+
+type Pool interface {
+	// Stop stops the worker pool. It does not wait to drain any internal queues. It must only be called once.
+	Stop()
```

Comment:
As far as I can tell, Stop is never called by the Flow controllers. It may be tricky to solve this one, since currently pools are shared between modules and the root controller. 

But also because there's currently only one Pool, it might not be a big deal that it leaks, since the root controller persists through the lifetime of the process.